dial_queue.go

```diff
@@ -0,0 +1,252 @@
+package dht
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-peer"
+	"github.com/libp2p/go-libp2p-peerstore/queue"
```

Comment:
Could you name the imports? Otherwise it's not obvious which identifier they end up with.

Reply:
Sure, this is not the official PR, just a venue to discuss the solution, but good call!

Reply:
✅